From b62b038cdf58c54205987dfc52ebf228505ad03b Mon Sep 17 00:00:00 2001
From: Etoh <etoh@syncplay.pl>
Date: Tue, 11 Apr 2023 20:22:52 +0100
Subject: [PATCH] Fix ArchLinux GUI TypeError (#600)

---
 syncplay/ui/gui.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/syncplay/ui/gui.py b/syncplay/ui/gui.py
index 7c100542..ca0f5d4a 100755
--- a/syncplay/ui/gui.py
+++ b/syncplay/ui/gui.py
@@ -2106,7 +2106,10 @@ def __init__(self, passedBar=None):
         if isMacOS():
             self.setWindowFlags(self.windowFlags())
         else:
-            self.setWindowFlags(self.windowFlags() & Qt.AA_DontUseNativeMenuBar)
+            try:    
+            	self.setWindowFlags(self.windowFlags() & Qt.AA_DontUseNativeMenuBar)
+            except TypeError:
+            	self.setWindowFlags(self.windowFlags())
         self.setWindowTitle("Syncplay v" + version + revision)
         self.mainLayout = QtWidgets.QVBoxLayout()
         self.addTopLayout(self)
